app/components/Account.js

```diff
@@ -11,53 +11,81 @@ export default class Account extends Component {
   @observable cryptoHillsBalance = new BigNumber(0);
 
   async componentDidMount() {
-    this.hillCoreInstance = await this.props.store.HillCore.deployed();
     this.auction = await this.props.store.AuctionBase.deployed();
     window.s = this;
-    this.getCryptoHillsBalance();
-    const blockWatcher = observe(this.props.store, "currentBlock", change => {
-      this.getCryptoHillsBalance();
-    });
-    const balanceWatcher = observe(this, "cryptoHillsBalance", change => {
+    this.hillCoreWatcher = when(
+      () => this.props.store.hillCoreInstance,
+      () => {
+        this.hillCoreWatcher = null;
+        this.blockWatcher = observe(
+          this.props.store,
+          "currentBlock",
+          change => {
+            this.getCryptoHillsBalance();
+          }
+        );
+      }
+    );
+
+    this.balanceWatcher = observe(this, "cryptoHillsBalance", change => {
       this.getCryptoHills();
     });
   }
 
+  componentWillUnmount() {
+    if (this.hillCoreWatcher) {
+      this.hillCoreWatcher();
```

Comment:
Does just calling them like this cancel them?

Reply:
Yeah, `watch` and `autorun` return disposer functions that you can call in `componentWillUnmount`